pkg/schedule/schedule.go

```diff
@@ -17,15 +17,23 @@ package schedule
 import (
 	"context"
 	"sync"
+
+	"go.etcd.io/etcd/client/pkg/v3/verify"
+
+	"go.uber.org/zap"
 )
 
-type Job func(context.Context)
+type Job struct {
```

Comment:
no strong opinion, but probably I would make it an interface... 

and have a:
   `NewJob("foo", func() {...} )` constructor...
   `NewRecoveringJob("boo", func() {...}, func() { ... } );

Reply:
Thanks for your advice.